hedera-node/cli-clients/src/main/java/com/hedera/services/cli/sign/RecordStreamSigningUtils.java

```diff
@@ -0,0 +1,287 @@
+                    final byte[] startRunningHash =
+                            f.get().getStartObjectRunningHash().getHash().toByteArray();
+                    final byte[] endRunningHash =
+                            f.get().getEndObjectRunningHash().getHash().toByteArray();
+                    final int version = recordResult.getKey();
+                    final byte[] serializedBytes = f.get().toByteArray();
+
+                    // update stream digest
+                    for (final int value : fileHeader) {
+                        dosMeta.writeInt(value);
+                    }
+                    dosMeta.write(startRunningHash);
+                    dosMeta.write(endRunningHash);
+                    dosMeta.writeLong(blockNumber);
+                    dosMeta.flush();
+
+                    // update meta digest
+                    dos.writeInt(version);
+                    dos.write(serializedBytes);
+                    dos.flush();
+                } else {
+                    System.err.printf(
+                            "outputStreamDigest :: Failed to parse record file [%s] with exception : [%s]%n",
+                            recordFile, "RecordStreamFile does not have startObjectRunningHash");
+                    throw new IOException("RecordStreamFile does not have startObjectRunningHash");
+                }
+            } else {
+                System.err.printf(
+                        "outputStreamDigest :: Failed to parse record file [%s] with exception : [%s]%n",
+                        recordFile, "RecordStreamFile is empty");
+                throw new IOException("RecordStreamFile is empty");
+            }
+
+        } catch (final IOException e) {
+            System.err.printf(
+                    "outputStreamDigest :: Got IOException when output steam digest [%s] with exception : [%s]%n",
+                    recordFile, e);
+            throw e;
+        }
+    }
+
+    private static void initRecordDigest() throws NoSuchAlgorithmException {
+        try {
+            streamDigest = MessageDigest.getInstance(DigestType.SHA_384.algorithmName());
+            metadataStreamDigest = MessageDigest.getInstance(DigestType.SHA_384.algorithmName());
+        } catch (final NoSuchAlgorithmException e) {
+            System.err.printf("initRecordDigest :: Failed to get message digest with exception : [%s]%n", e);
+            throw e;
+        }
+    }
+
+    private static SignatureObject generateSignatureObject(final byte[] hash, final KeyPair sigKeyPair)
+            throws NoSuchAlgorithmException, SignatureException, NoSuchProviderException, InvalidKeyException {
+        final byte[] signature = signData(hash, sigKeyPair);
+        return SignatureObject.newBuilder()
+                .setType(SHA_384_WITH_RSA)
+                .setLength(signature.length)
+                .setChecksum(101 - signature.length) // simple checksum to detect if at wrong place in
```

Comment:
This must be part of the spec of a signature file, but where?  It's the `101-length` I'm curious about - seems random.  A pointer to some kind of spec/other defining code would be nice.

Reply:
This part is the refactored code. I don't know any spec about it either.

Reply:
101 is used in the original RecordStreamFileWriter.java  

```java
    private SignatureObject generateSignatureObject(final byte[] hash) {
        final var signature = signer.sign(hash).getSignatureBytes();
        return SignatureObject.newBuilder()
                .setType(SignatureType.SHA_384_WITH_RSA)
                .setLength(signature.length)
                .setChecksum(101 - signature.length) // simple checksum to detect if at wrong place in
                // the stream
                .setSignature(ByteStringUtils.wrapUnsafely(signature))
                .setHashObject(toProto(hash))
                .build();
    }
```

Reply:
Thanks @JeffreyDallas